x-pack/plugins/monitoring/server/telemetry_collection/get_logstash_stats.ts

```diff
@@ -393,20 +468,54 @@ export async function getLogstashStats(
     plugins: {},
   };
 
-  await fetchLogstashStats(callCluster, clusterUuids, start, end, options);
-  await Promise.all(
-    clusterUuids.map(async (clusterUuid) => {
-      if (options.clusters[clusterUuid] !== undefined) {
-        await fetchLogstashState(
-          callCluster,
-          clusterUuid,
-          options.allEphemeralIds[clusterUuid],
-          start,
-          end,
-          options
-        );
-      }
-    })
-  );
+  // if index name contains '-mb', means metricbeat based monitoring
+  // filter_path and collapse fields in the queries differ on metricbeat vs. self-monitoring
+  // note: agent driven LS monitoring indices pattern differ ".ds-metrics-logstash*"
+  for (const clusterUuid of clusterUuids) {
+    const isSelfMonitoring: boolean = await isLogstashSelfMonitoring(callCluster, clusterUuid);
+
+    await fetchLogstashStats(callCluster, clusterUuids, start, end, options, isSelfMonitoring);
+
+    if (options.clusters[clusterUuid] !== undefined) {
+      await fetchLogstashState(
+        callCluster,
+        clusterUuid,
+        options.allEphemeralIds[clusterUuid],
+        start,
+        end,
+        options,
+        isSelfMonitoring
+      );
+    }
+  }
   return options.clusters;
 }
+
+export async function isLogstashSelfMonitoring(
```

Comment:
review note: I do see internal monitoring has a [`metricbeatUsed` param](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/server/kibana_monitoring/collectors/lib/fetch_stack_product_usage.ts#L100) but as I have checked 1) it doesn't get true when I tested with metricbeat but I don't have any ideas how UI is displaying properly; 2) there seem no way to utilize it since collection types differ.